common/models/user.js

```diff
@@ -673,6 +673,48 @@ module.exports = function(User) {
       next();
     });
 
+    // Delete old sessions once email is updated
+    UserModel.observe('before save', function beforeEmailUpdate(ctx, next) {
+      if (ctx.isNewInstance) return next();
+      if (ctx.where) {
+        ctx.Model.find({ where: ctx.where }, function(err, userInstances) {
+          if (err) return next(err);
+          getOriginalUserData(userInstances);
+        });
+      } else if (ctx.instance) {
+        ctx.hookState.id = ctx.instance.id;
+        var oldId = ctx.hookState.id;
+        ctx.Model.find({ where: { id: oldId }}, function(err, userInstances) {
+          if (err) return next (err);
+          getOriginalUserData(userInstances);
+        });
+      }
+
+      function getOriginalUserData(userInstances) {
+        ctx.hookState.originalUserData = userInstances.map(function(u) {
+          return { id: u.id, email: u.email };
+        });
+      }
+
+      UserModel.observe('after save', function afterEmailUpdate(ctx, next) {
+        var AccessToken = ctx.Model.relations.accessTokens.modelTo;
+        var  newEmail = (ctx.instance || ctx.data).email;
+        if (!ctx.hookState.originalUserData) return next();
+        var idsToExpire = ctx.hookState.originalUserData.filter(function(u) {
+          return u.email !== newEmail;
+        }).map(function(u) {
+          return u.id;
+        });
+        if (!idsToExpire.length) return next();
+        AccessToken.deleteAll({ userId: { inq: idsToExpire }}, function(err, info) {
+          if (err) return next(err);
+        });
+        next();
+      });
+
+      next();
```

Comment:
`next()` in "before save" hook must be called only after all async operations finished and we have all required data ready.